File: myungjoo/0x06/10845.cpp
#include<iostream>
#include<queue>
#include<string>

using namespace std;

int main(void) {
    int N;
    cin >> N;
    queue<int> q;
    int num;

    for (int i=0;i<N;i++){
        string command;
        cin >> command;

        if(command == "push"){
            int data;
            cin >> data;
            q.push(data);
        }
        else if (command == "pop"){
            if (q.size()!=0){
                num = q.front();
                q.pop();
            }
            else{
                num = -1;
            }
            cout << num << endl;
            }
        else if (command == "size"){
            cout << q.size() << endl;
        }
        else if (command == "empty"){
            if (q.size()==0) num = 1;
            else num = 0;
            cout << num << endl;
            }
        else if (command == "front"){
            if(q.size() == 0) num = -1;
            else num = q.front();
            cout << num << endl;
        }
        else if (command == "back"){
            if(q.size() == 0) num = -1;
            else num = q.back();
            cout << num << endl;
        }
        }

    return 0;
}

File: myungjoo/0x06/10845.py

import sys

N = int(sys.stdin.readline())
queue = []

for i in range(N):
    command = sys.stdin.readline().split()
    
    if command[0] == "push":
        queue.insert(0, command[1]) #맨 앞에 넣어야함
    elif command[0] == "pop":
        if len(queue) == 0:
            print(-1)
        else :
            print(queue.pop())
    elif command[0] == "size":
        print(len(queue))
    elif command[0] == "empty":
        if len(queue) == 0:
            print(1)
        else: print(0)
    elif command[0] == "front":
        if len(queue) == 0:
            print(-1)
        else: 
            print(queue[-1])
    elif command[0] == "back":
        if len(queue) == 0:
            print(-1)
        else: 
            print(queue[0])
        